.eslintrc.js

```diff
@@ -0,0 +1,54 @@
+module.exports = {
+  root: true,
+  parser: '@typescript-eslint/parser',
+  // The following based on
+  // https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/TYPED_LINTING.md.
+  //
+  parserOptions: {
+    tsconfigRootDir: __dirname,
+    project: ['./tsconfig.json'],
+  },
+  extends: [
+    'eslint:recommended',
+    'plugin:@typescript-eslint/eslint-recommended',
+    'plugin:@typescript-eslint/recommended',
+    // Note: `recommended-requiring-type-checking` have type-aware rules. This comes
+    // with a performance penalty. For small projects, this is usually negligible.
+    // It is recommended to separate the linting into two stagings once the type-aware
+    // checking becomes a productivity issue. See second half of the "Recommended
+    // Configs" section at
+    // https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin#recommended-configs.
+    'plugin:@typescript-eslint/recommended-requiring-type-checking',
+    'plugin:react/recommended',
+  ],
+  plugins: [
+    '@typescript-eslint',
+  ],
+  rules: {
+    'brace-style': ['error', 'stroustrup', { allowSingleLine: true }],
+    camelcase: ['error', { allow: ['_unused$'] }],
+    'comma-dangle': ['error', {
+      arrays: 'always-multiline',
+      exports: 'always-multiline',
+      functions: 'ignore',
+      imports: 'always-multiline',
+      objects: 'always-multiline',
+    }],
+    'no-multiple-empty-lines': ['error', { max: 2, maxBOF: 1, maxEOF: 1 }],
+    'padded-blocks': ['off'],
+
+    // Off because we are using TypeScript which expects us to declare the props.
+    'react/prop-types': 'off',
+
+    '@typescript-eslint/explicit-function-return-type': 'off',
+
+    // The standard `no-unused-vars` rule is handling unused variables.
+    // Off because it incorrectly detects whether a variable is unused.
+    // '@typescript-eslint/no-unused-vars': 'off',
```

Comment:
Following [this doc](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unused-vars.md), this seems to work for me:

```suggestion
    // Disabling the base rule as it can report incorrect errors
    'no-unused-vars': 'off',
    '@typescript-eslint/no-unused-vars': ['error'],
```

I ran `npm run lint:js` and got no errors with this change.

Reply:
Nice find. I am really surprised that the base rule is not disabled by default since we are extending `plugin:@typescript-eslint/eslint-recommended` and `plugin:@typescript-eslint/recommended`.

@yanir I have started an ordering convention in `.eslintrc.js`. I don't think there is a rule we can use to help keep it in order or even if the team is in agreement. Basically, I have the ESLint rules grouped by whether it is a base, built-in rule, from `eslint-plugin-react` (the rule names begin with `react/`, or from `@typescript-eslint/eslint-plugin` (the rule names begin with `@typescript-eslint/`). Then within each group, they are alphabetized by rule name. This just makes it easier to find things. Once we pretty much finalize the rules, we won't be make many, if any, changes.

Reply:
Ordering sounds good! Can just move the `no-unused-vars` up with the comment.

I haven't dug too deep, but looks like they're working on a better solution: https://github.com/typescript-eslint/typescript-eslint/issues/1856